WGLMakie/src/imagelike.jl

```diff
@@ -163,26 +163,22 @@ function limits_to_uvmesh(plot, f32c)
     # can be displayed as a rectangle
     t = Makie.transform_func_obs(plot)[]
 
-    # TODO, this branch is only hit by Image, but not for Heatmap with stepranges
-    # because convert_arguments converts x/y to Vector{Float32}
-    if px[] isa Makie.Interval && py[] isa Makie.Interval && Makie.is_identity_transform(t)
+    if px[] isa Makie.EndPoints && py[] isa Makie.EndPoints && Makie.is_identity_transform(t)
         rect = lift(plot, px, py) do x, y
-            xmin, xmax = Makie.endpoints(x)
-            ymin, ymax = Makie.endpoints(y)
+            xmin, xmax = extrema(x)
+            ymin, ymax = extrema(y)
```

Comment:
Not sure if this is something to do here or in #3989, but a lot of these extrema probably need to be replaced with splatting or `x[1], x[2]` if we want to support reversed images, heatmaps etc

Reply:
Yeah, I wouldn't want to do it here, and sticking with extrema makes it easier to find once we get to it.